code/game/machinery/PDApainter.dm

```diff
@@ -182,6 +185,11 @@
 	cached_icon_state = icon2base64(pda_sprite)
 	SStgui.update_uis(src)
 
+/obj/machinery/pdapainter/proc/on_pda_qdel()
+	UnregisterSignal(storedpda, COMSIG_PARENT_QDELETING)
```

Comment:
add an early return here if `storedpda` is null, trying to use these signal procs on a null object will runtime